demos/projects/PC/linux/config/demo_config.h

```diff
@@ -0,0 +1,188 @@
+ * 
+ */
+#define democonfigID_SCOPE                  "<YOUR ID SCOPE HERE>"
+
+/**
+ * @brief Registration Id of provisioning service
+ * 
+ * @warning If using X509 authentication, this MUST match the Common Name of the cert.
+ *
+ *  @note https://docs.microsoft.com/en-us/azure/iot-dps/concepts-service#registration-id
+ */
+#define democonfigREGISTRATION_ID           "<YOUR REGISTRATION ID HERE>"
+
+#endif // democonfigENABLE_DPS_SAMPLE
+
+/**
+ * @brief IoTHub device Id.
+ *
+ */
+#define democonfigDEVICE_ID                 "<YOUR DEVICE ID HERE>"
+
+/**
+ * @brief IoTHub module Id.
+ *
+ */
+#define democonfigMODULE_ID                 ""
+/**
+ * @brief IoTHub hostname.
+ *
+ */
+#define democonfigHOSTNAME                  "<YOUR IOT HUB HOSTNAME HERE>"
+
+/**
+ * @brief Device symmetric key
+ *
+ */
+#define democonfigDEVICE_SYMMETRIC_KEY      "<Symmetric key>"
+
+/**
+ * @brief Client's X509 Certificate.
+ *
+ */
+// #define democonfigCLIENT_CERTIFICATE_PEM    "<YOUR DEVICE CERT HERE>"
+
+/**
+ * @brief Client's private key.
+ * 
+ */
+// #define democonfigCLIENT_PRIVATE_KEY_PEM    "<YOUR DEVICE PRIVATE KEY HERE>"
+
+/**
+ * @brief Baltimore Trusted RooT CA.
+ *
+ */
+#define democonfigROOT_CA_PEM "-----BEGIN CERTIFICATE-----\r\n" \
+"MIIDdzCCAl+gAwIBAgIEAgAAuTANBgkqhkiG9w0BAQUFADBaMQswCQYDVQQGEwJJ\r\n" \
+"RTESMBAGA1UEChMJQmFsdGltb3JlMRMwEQYDVQQLEwpDeWJlclRydXN0MSIwIAYD\r\n" \
+"VQQDExlCYWx0aW1vcmUgQ3liZXJUcnVzdCBSb290MB4XDTAwMDUxMjE4NDYwMFoX\r\n" \
+"DTI1MDUxMjIzNTkwMFowWjELMAkGA1UEBhMCSUUxEjAQBgNVBAoTCUJhbHRpbW9y\r\n" \
+"ZTETMBEGA1UECxMKQ3liZXJUcnVzdDEiMCAGA1UEAxMZQmFsdGltb3JlIEN5YmVy\r\n" \
+"VHJ1c3QgUm9vdDCCASIwDQYJKoZIhvcNAQEBBQADggEPADCCAQoCggEBAKMEuyKr\r\n" \
+"mD1X6CZymrV51Cni4eiVgLGw41uOKymaZN+hXe2wCQVt2yguzmKiYv60iNoS6zjr\r\n" \
+"IZ3AQSsBUnuId9Mcj8e6uYi1agnnc+gRQKfRzMpijS3ljwumUNKoUMMo6vWrJYeK\r\n" \
+"mpYcqWe4PwzV9/lSEy/CG9VwcPCPwBLKBsua4dnKM3p31vjsufFoREJIE9LAwqSu\r\n" \
+"XmD+tqYF/LTdB1kC1FkYmGP1pWPgkAx9XbIGevOF6uvUA65ehD5f/xXtabz5OTZy\r\n" \
+"dc93Uk3zyZAsuT3lySNTPx8kmCFcB5kpvcY67Oduhjprl3RjM71oGDHweI12v/ye\r\n" \
+"jl0qhqdNkNwnGjkCAwEAAaNFMEMwHQYDVR0OBBYEFOWdWTCCR1jMrPoIVDaGezq1\r\n" \
+"BE3wMBIGA1UdEwEB/wQIMAYBAf8CAQMwDgYDVR0PAQH/BAQDAgEGMA0GCSqGSIb3\r\n" \
+"DQEBBQUAA4IBAQCFDF2O5G9RaEIFoN27TyclhAO992T9Ldcw46QQF+vaKSm2eT92\r\n" \
+"9hkTI7gQCvlYpNRhcL0EYWoSihfVCr3FvDB81ukMJY2GQE/szKN+OMY3EU/t3Wgx\r\n" \
+"jkzSswF07r51XgdIGn9w/xZchMB5hbgF/X++ZRGjD8ACtPhSNzkE1akxehi/oCr0\r\n" \
+"Epn3o0WC4zxe9Z2etciefC7IpJ5OCBRLbf1wbWsaY71k5h+3zvDyny67G7fyUIhz\r\n" \
+"ksLi4xaNmjICq44Y3ekQEe5+NauQrz4wlHrQMz2nZQ/1/I6eYs9HRCwBXbsdtTLS\r\n" \
+"R9I4LtD+gdwyah617jzV/OeBHRnDJELqYzmp\r\n" \
+"-----END CERTIFICATE-----\r\n"
+
+/**
+ * @brief An option to disable Server Name Indication.
+ *
+ * @note When using a local Mosquitto server setup, SNI needs to be disabled
+ * for an MQTT broker that only has an IP address but no hostname. However,
+ * SNI should be enabled whenever possible.
+ */
+#define democonfigDISABLE_SNI              ( pdFALSE )
```

Comment:
Ugh - in perfect world we wouldn't have this type of flag that user flips on to get things going and then forgets it and ships.

Could we put some sort of thing in main() where like

```c
#if (democonfigDISABLE_SNI == pdTrue)
   LogWarn("You've disabled SNI for this test.  Some sort of properly word-smithed callout on this");  
#endif
```

Reply:
SNI should not be disable, I will remove this option and make it always true in sample